### DISCUSSION TITLE
How to save and load diarization results ?

### DISCUSSION THREAD

**Discussion by amitli1** [pyannote/pyannote-audio#1341] [Development]
I want to save the `pyannote/speaker-diarization `results to file (text or pickle) and load it later (instead of rerunning the diarization pipeline).

Is it possible ?

```
pipeline   = Pipeline.from_pretrained("pyannote/speaker-diarization")
diarization = pipeline("test.wav")
```

I want to save `diarization`

I saw that I can save it as `rttm`  file, but don't have the opportunity to load this `rttm` to `pyannote.core.annotation.Annotation`

**Comment by diagomike** [comment]
this is how you write to rttm file 
```python
with open('out_put.rttm','w') as f:
  groundtruth.write_rttm(f)
```

**Reply by rmatinha-fctunl** [reply]
It says that the 'DiarizeOutput' object has no attribute 'write_rttm'

**Reply by hbredin** [reply]
diarization.speaker_diarization.write_rttm(...)

**Reply by rmatinha-fctunl** [reply]
Thank you so much! It worked! I had opened a discussion with a similar problem before, do you mind if I paste this solution there?

**Comment by diagomike** [comment]
this will save `index  [start -> end]` format into a simple text, which you can easily write to disk
```python
simple_text = ''
for (start,end),index in groundtruth.itertracks():
  simple_text += f"%s: [%.2fs -> %.2fs]\n" % (index,start, end)
```

**Comment by mcho421** [comment]
It looks like there's support to read and write .rttm or .lab files. Here's an example using .lab: 

```
# Writing to a file on disk
diarization.write_lab('result.lab')

# Writing to a string
result = diarization.to_lab()
print(result)

# Loading the file
from pyannote.database.util import load_lab
diarization = load_lab("result.lab")
```
